pkg/kube/client_generator.go

```diff
@@ -0,0 +1,1024 @@
+		}
+	}()
+	if err != nil {
+		return errors.Wrapf(err, "creating tempfile to compile %s to %v", SchemaWriterSrcFileName, err)
+	}
+	cmd := util.Command{
+		Dir:  inputBase,
+		Name: "go",
+		Args: []string{
+			"build",
+			"-o",
+			schemaWriterBinary.Name(),
+			schemaWriterSrc,
+		},
+	}
+	_, err = cmd.RunWithoutRetry()
+	if err != nil {
+		return errors.Wrapf(err, "running %s", cmd.String())
+	}
+	fileJSON := filepath.Join(outputDir, OpenApiV2JSON)
+	fileYAML := filepath.Join(outputDir, OpenApiV2YAML)
+	cmd = util.Command{
+		Name: schemaWriterBinary.Name(),
+		Args: []string{
+			"--output-directory",
+			outputDir,
+			"--title",
+			title,
+			"--version",
+			version,
+		},
+	}
+	_, err = cmd.RunWithoutRetry()
+	if err != nil {
+		return errors.Wrapf(err, "running %s", cmd.String())
+	}
+	// Convert to YAML as well
+	bytes, err := ioutil.ReadFile(fileJSON)
+	if err != nil {
+		return errors.Wrapf(err, "reading %s", fileJSON)
+	}
+	yamlBytes, err := yaml.JSONToYAML(bytes)
+	if err != nil {
+		return errors.Wrapf(err, "converting %s to yaml", fileJSON)
+	}
+	err = ioutil.WriteFile(fileYAML, yamlBytes, 0755)
+	if err != nil {
+		return errors.Wrapf(err, "writing %s", fileYAML)
+	}
+	return nil
+}
+
+// InstallGenApiDocs installs the gen-apidocs tool from the kubernetes-incubator/reference-docs repository.
+// It takes a version (really a commit-ish) and the gitter client
+func InstallGenApiDocs(version string, gitter gits.Gitter) (string, error) {
+	binDir, err := util.JXBinLocation()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting ~/.jx/bin")
+	}
+	log.Infof("Installing %s %s to %s\n", apiDocsGeneratorGitUrl, version, binDir)
+	dir, err := cloneOrFetch(apiDocsGeneratorGitUrl, version, gitter)
+	if err != nil {
+		return "", err
+	}
+
+	installCmd := util.Command{
+		Dir:  dir,
+		Name: "go",
+		Args: []string{
+			"build",
+			"-o",
+			filepath.Join(binDir, apiDocsGenerator),
+			fmt.Sprintf("%s/main.go", apiDocsGeneratorDir),
+		},
+	}
+	_, err = installCmd.RunWithoutRetry()
+	if err != nil {
+		return "", errors.Wrapf(err, "running %s", installCmd.String())
+	}
+	return dir, nil
+}
+
+// AssembleApiDocsStatic copies the static files from the referenceDocsRepo to the outputDir.
+// It also downloads from CDN jquery and bootstrap js
+func AssembleApiDocsStatic(referenceDocsRepo string, outputDir string) error {
+	srcDir := filepath.Join(referenceDocsRepo, "gen-apidocs", "generators", "static")
+	outDir := filepath.Join(outputDir, "static")
+	if empty, err := util.IsEmpty(outDir); err != nil {
+		return errors.Wrapf(err, "checking %s is empty", outDir)
+	} else if empty {
+		log.Infof("Copying static files from %s to %s\n", srcDir, outDir)
+		err := util.CopyDir(srcDir, outDir, false)
+		if err != nil {
+			return errors.Wrapf(err, "copying %s to %s", srcDir, outDir)
+		}
+		err = downloadFile("https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js", filepath.Join(outDir,
```

Comment:
Maybe extract these in a const. It would easier to maintain them when it needs a version update.